### PR TITLE
libutils: trace: support more than 100 threads and cores

### PR DESCRIPTION
**Pull request by jforissier**
The first patch is a fix that is useless if the second patch is applied, so perhaps I should drop it?

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
> The first patch is a fix that is useless if the second patch is applied, so perhaps I should drop it?

Doesn't matter to me.
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

File: lib/libutils/ext/trace.c [hunk 1]
```diff
@@ -73,16 +75,21 @@ static int print_thread_id(char *buf, size_t bs)
 #if defined(__KERNEL__)
 static int print_core_id(char *buf, size_t bs)
 {
-#if CFG_TEE_CORE_NB_CORE > 10
+#if CFG_TEE_CORE_NB_CORE > 100
+	const int num_digits = 3;
+	const char *qm = "???";
```
[the remark this answers is not in the document]
Comment (jforissier) [reply lib/libutils/ext/trace.c:80]: No big difference I suppose?

[the remark this answers is not in the document]
Comment (etienne-lms) [reply lib/libutils/ext/trace.c:80]: For consistency in optee_os implementation, maybe prefer the later?
That's nitpicking. Anyhow, `Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` for both commits.

[the remark this answers is not in the document]
Comment (jforissier) [reply lib/libutils/ext/trace.c:80]: OK, changed, I don't have a strong preference either way.

**Pull request merged**